runtime/compiler/control/J9Options.cpp

```diff
@@ -1919,6 +2049,110 @@ J9::Options::fePreProcess(void * base)
+            if (xxUseJITServerArgIndex >= 0)
+               {
+               // retrieve whole option part, i.e., server=<address/hostname>,port=<number>
+               char *options = NULL;
+               GET_OPTION_OPTION(xxUseJITServerArgIndex, ':', ':', &options);
+
+               char *scanLimit = options + strlen(options);
+               char delimiter = ',';
+               char *unRecognizedOptions = NULL;
+               while (options < scanLimit)
+                  {
+                  if (try_scan(&options, "server=")) // parse server=<address/hostname> option
+                     {
+                     char *address = scan_to_delim(PORTLIB, &options, delimiter);
+                     compInfo->getPersistentInfo()->setJITServerAddress(address);
+                     j9mem_free_memory(address);
+                     }
+                  else if (!JITServerParseOptionsCommon(&options, delimiter, compInfo, &unRecognizedOptions))
+                     break;
+                  }
+               }
+            }
+         else // server mode
+            {
+            compInfo->getPersistentInfo()->setRemoteCompilationMode(JITServer::SERVER);
+
+            // parse -XX:+StartAsJITServer:port=<number> option if provided
+            char *xxStartAsJITServerOptionWithArgs = "-XX:+StartAsJITServer:"; // tail colon indicates options are provided
+            xxStartAsJITServerArgIndex = FIND_ARG_IN_VMARGS(STARTSWITH_MATCH, xxStartAsJITServerOptionWithArgs, 0);
+            if (xxStartAsJITServerArgIndex >= 0)
+               {
+               // retrieve whole option part, i.e., port=<number>
+               char *options = NULL;
+               GET_OPTION_OPTION(xxStartAsJITServerArgIndex, ':', ':', &options);
+
+               char *scanLimit = options + strlen(options);
+               char delimiter = ',';
+               char *unRecognizedOptions = NULL;
+               while (options < scanLimit && JITServerParseOptionsCommon(&options, delimiter, compInfo, &unRecognizedOptions));
+               }
+            }
+         }
+      if (compInfo->getPersistentInfo()->getRemoteCompilationMode() != JITServer::NONE)
+         {
+         // generate a random identifier for this JITServer instance.
+         // TODO: prevent collisions with some kind of atomic registration algo!
+         std::random_device rd;
+         std::mt19937_64 rng(rd());
+         std::uniform_int_distribution<uint64_t> dist;
+         compInfo->getPersistentInfo()->setClientUID(dist(rng));
+         }
+      }
+   // Set a value for _safeReservePhysicalMemoryValue that is proportional
+   // to the amount of free physical memory at boot time
+   // For JITServer we can use a 0 value because compilations are done remotely
+   // The user can still override it with a command line option
+   if (compInfo->getPersistentInfo()->getRemoteCompilationMode() == JITServer::CLIENT)
```

Comment:
Looks like the `else` part might need to be removed since `J9::Options::_safeReservePhysicalMemoryValue` is set at the following line which is before `JITServer` related options are parsed. 

https://github.com/eclipse/openj9/blob/1fcc869f35898d78652a9f1e07e53a40280d9110/runtime/compiler/control/J9Options.cpp#L1890

Reply:
ok. I think I moved that part of the code down here, so that I can have access to JITServer information.
You can delete the else part.